### DISCUSSION TITLE
Why am I not being given a reference to the class in a class decorator function?

### DISCUSSION THREAD

**Discussion by matthewadams** [babel/babel#12160] [Q&A] [answered]
All details are at https://github.com/tc39/proposal-decorators/issues/339.

It seems to me that a class decorator function should be given a reference to the class itself.  Am I missing something fundamental here?

**Comment by nicolo-ribaudo** [comment, marked as answer]
If anyone else is looking for the answer for this question, please read https://github.com/tc39/proposal-decorators/issues/339!
